src/heuristic/HeuristicMapper.cpp

```diff
@@ -306,6 +339,7 @@ void HeuristicMapper::mapUnmappedGates(
     std::size_t layer, std::vector<std::uint16_t>& consideredQubits) {
   for (const auto& gate : layers.at(layer)) {
     if (gate.singleQubit()) {
+      // TODO: if considerFidelity probably necessary to also map single qubits
```

Comment:
Do you plan to address this Todo within this PR or would you rather prefer to do that as a follow up?

Reply:
I think I would tackle the lookahead problem next, as I feel the interaction between layers is even more integral.
Only after that am I planning on integrating the noise-aware approach with dynamic initial layouts.
How I should split those into individual PRs is completely up to you, though. Before and after each of these steps, the mapping should at least be working. However, I think the full potential of the noise-aware approach can only be utilized once lookahead and dynamic initial layouts are both implemented. I don't know which of those two is a sufficient or preferable state for integration into the main branch.

Reply:
I don't think I have a real preference for the particular steps towards achieving the grand goal of noise-aware mapping.
Although I agree that a working mapper should result from all individual contributions. Other than that, I typically prefer small PRs to huge ones, so splitting the whole contribution up would make sense.
Feel free to proceed as you see fit. 
In the end, I'll cut a release once everything is out.